core/blockchain.go

```diff
@@ -1333,6 +1343,32 @@ func (bc *BlockChain) writeBlockWithState(block *types.Block, receipts []*types.
 	return status, nil
 }
 
+// Fill blockResult content
+func (bc *BlockChain) writeBlockResult(state *state.StateDB, block *types.Block, blockResult *types.BlockResult) {
+	blockResult.TraceBlock = types.NewTraceBlock(bc.chainConfig, block)
+	for i, tx := range block.Transactions() {
+		evmTrace := blockResult.ExecutionResults[i]
+		// Get the sender's address.
+		from, _ := types.Sender(types.MakeSigner(bc.chainConfig, block.Number()), tx)
+		// Get account's proof.
+		proof, err := state.GetProof(from)
+		if err != nil {
+			log.Error("Failed to get proof", "blockNumber", block.NumberU64(), "address", from.String(), "err", err)
+		} else {
+			evmTrace.Proof = make([]string, len(proof))
+			for i := range proof {
+				evmTrace.Proof[i] = hexutil.Encode(proof[i])
+			}
+		}
+		// Contract is called
+		if len(tx.Data()) != 0 && tx.To() != nil {
+			evmTrace.ByteCode = hexutil.Encode(state.GetCode(*tx.To()))
+		} else if tx.To() == nil { // Contract is created.
+			evmTrace.ByteCode = hexutil.Encode(tx.Data())
+		}
```

Comment:
do we forget to deal with "For Transfer: tx.to address’s code_hash" ?

Reply:
Yeah, it a little awkward, i need to think how to add gracefully.

Reply:
Tx has the tx.to address's code_hash when is a contract call.

Reply:
We actually also need `code_hash` for transfer (instead of "contract call")

but seems it's fixed and we can calculate it in  rust : https://eips.ethereum.org/EIPS/eip-1186